policy-controller/src/main.rs

```diff
@@ -159,3 +177,67 @@ async fn sigterm() {
         _ => future::pending().await,
     };
 }
+
+fn log_init(LogLevel(level): LogLevel, format: LogFormat) -> Result<()> {
+    let (filter, _level) = reload::Layer::new(EnvFilter::new(level));
+    let registry = tracing_subscriber::registry().with(filter);
+
+    let dispatch = match format {
+        LogFormat::Plain => registry
+            .with(tracing_subscriber::fmt::layer().event_format(tracing_subscriber::fmt::format()))
+            .into(),
+
+        LogFormat::Json => {
+            let event_fmt = tracing_subscriber::fmt::format()
+                // Configure the formatter to output JSON logs.
+                .json()
+                // Output the current span context as a JSON list.
+                .with_span_list(true)
+                // Don't output a field for the current span, since this
+                // would duplicate information already in the span list.
+                .with_current_span(false);
+
+            // Use the JSON event formatter and the JSON field formatter.
+            let fmt = tracing_subscriber::fmt::layer()
+                .event_format(event_fmt)
+                .fmt_fields(format::JsonFields::default());
+
+            registry.with(fmt).into()
+        }
+    };
+
+    tracing::dispatcher::set_global_default(dispatch)?;
+
+    // Setup compatibility with the `log` crate (for dependencies) and ensure it honors our
+    // configured log level.
+    tracing_log::LogTracer::init()?;
+    tracing_log::log::set_max_level(tracing_log::AsLog::as_log(
+        &tracing::level_filters::LevelFilter::current(),
+    ));
```

Comment:
nit/TIOLI: if we use the `.init()` method on `Registry` rather than calling `dispatcher::set_global_default`, and enable the "tracing-log" feature flag on the `tracing-subscriber` crate, it will also set up the log compatibility for us automagically...but, we might prefer to do this explicitly rather than by magic via the feature flag?

Reply:
@olix0r opened #6895 for this change